ts-tests/tests/test-eth-balance.ts

```diff
@@ -68,7 +84,68 @@ async function eth_link(api: ApiPromise, alice: KeyringPair) {
 
   console.log(`\nStep 1: Link Ethereum account`);
 
-  const transaction = api.tx.accountLinkerModule.link(alice.address, 0, 0, 0, test_r, test_s, test_v);
+  const registry = new TypeRegistry();
+
+  // Encode prefix with concatenated utf8, instead of SCALE codec to match the litentry node implementation
+	let encodedPrefix = Buffer.from(msgPrefix, 'utf-8');
+  
+  //let encodedAccId = registry.createType('AccountId', alice.address).toU8a();
+  //console.log(encodedAccId);
+  //console.log(alice.addressRaw);
+
+  let encodedExpiredBlock = new UInt(registry, 10000, 32).toU8a();
+
+  let encodedMsg = new Uint8Array(encodedPrefix.length + alice.addressRaw.length + encodedExpiredBlock.length);
+  encodedMsg.set(encodedPrefix);
+  encodedMsg.set(alice.addressRaw, encodedPrefix.length);
+  encodedMsg.set(encodedExpiredBlock, encodedPrefix.length + alice.addressRaw.length);
+
+  // To use manual hash and sign method, a prefix of \x19Ethereum ... is also needed to be prefixed manually
+	//let hash = Hash.keccak256s(encodedMsg);
+
+  //console.log('hash is:');
+  //console.log(hash);
+
+	// TODO ECDSA keyring from polkadot crypto still not working
+  //const ropstenTestAcc = keyringRopsten.addFromUri('0xe82c0c4259710bb0d6cf9f9e8d0ad73419c1278a14d375e5ca691e7618103011');
+  //console.log('ropsten pub key: ');
+  //console.log(ropstenTestAcc.publicKey);
+  //console.log(ropstenTestAcc.address);
+  //console.log(crypto.keccakAsU8a(ropstenTestAcc.publicKey));
+  //let signedMsg = ropstenTestAcc.sign(new Buffer(hash2.slice(2), "hex"));
+  //let r = signedMsg.slice(0, 32);
+  //let s = signedMsg.slice(32, 64);
+  //let v = signedMsg[64];
+	
+  //var signature = Account.sign(hash, privateKey);
+  //var vrs = Account.decodeSignature(signature);
+  //console.log("signature is :");
+  //console.log(signature);
+  //console.log(keyPair.sign(new Buffer(hash2.slice(2), "hex"), { canonical: true }).r.toString(16));
+
+	// TODO Web3 could be used to replace eth-lib once ethereum prefix is implemented on account-linker side
+	const Web3 = require("web3");
+  const web3 = new Web3(
+    "wss://ropsten.infura.io/ws/v3/aa0a6af5f94549928307febe80612a2a"
```

Comment:
Several issues here:
1. Because sign can be done offline, this test should not depend on infura which may introduce network error.
2. Nothing to do with ropsten
3. the api key is here (this is a public repo)

Reply:
Absolutely right. The initial implementation here was with use of elliptic lib, which is also the library used by web3js. The reason to replace with web3js here is to simulate the behavior that most ethereum clients do about how to sign a message. 

Nevertheless it's using web3, but network is not being used. Instead it's just signing a message with a given private key. I will try to remove network dependency here if possible. Otherwise I will rewrite here with elliptic crypto lib instead.